android/app/src/main/java/com/couchbase/todo/db/DumpTask.java

```diff
@@ -0,0 +1,44 @@
+package com.couchbase.todo.db;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import com.couchbase.lite.CouchbaseLiteException;
+import com.couchbase.lite.Document;
+import com.couchbase.lite.Expression;
+import com.couchbase.lite.Meta;
+import com.couchbase.lite.Ordering;
+import com.couchbase.lite.Query;
+import com.couchbase.lite.Result;
+import com.couchbase.lite.ResultSet;
+import com.couchbase.lite.SelectResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public final class DumpTask extends AsyncTask<String,Void,Void> {
+    //private volatile Consumer<List<Document>> listener;
+   //public FetchTask(Consumer<List<Document>> listener) { this.listener = listener; }
+    @Override
+    protected Void doInBackground(String... ids) {
+        //dump each task/each list
+        for (String id : ids) {
+            Document doc = DAO.get().fetch(id);
+            Log.i("doc",doc.toJSON());
```

Comment:
Ohhhhhh!!! it took me a while to see what you are doing here.

Let's talk about this one:  If you try to describe what this method does, it is going to be complicated and involve lots of "if"s.  I think that, perhaps, it should just log the list of docs it is given.  I might be convinced otherwise, tho...